File: src/main/java/com/tienda/proyectofinal/model/Cita.java
package com.tienda.proyectofinal.model;


import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import jakarta.validation.constraints.*;

@Entity
@Table(name = "citas")
@Data
public class Cita {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "id_usuario")
    private Usuario usuario;

    @NotNull
    private LocalDateTime fecha;

    @ManyToOne(optional = false)
    @JoinColumn(name = "servicio", referencedColumnName = "nombre")
    private Servicio servicio;


}

File: app/src/main/java/com/example/proyectofinalandroid/Cita.kt
package com.example.proyectofinalandroid
import java.io.Serializable

data class Cita(
    val id: Int? =null,
    val usuario: Usuario,
    val fecha: String, // formato: "2025-06-01T15:30:00"
    val servicio: Servicio
) :Serializable

